ir/type.h

```diff
@@ -17,6 +17,7 @@ class VoidType;
 class Type {
 protected:
   std::string name;
+  bool defined;
```

Comment:
as noted in the review of the parser, I don't think this change is needed.

Reply:
i though it would be more cleaner and efficient this way; 

1) I didn't want to use `dynamic_cast<SymbolicType>` to check if `parse_type` actually parsed something or not.

2) We were using `defined` in `IntType` class anyways and there was no way to check if any `SymbolicType` is defined or not. 

So, hoisting `defined` to the base class seemed like a good choice to me. However, your other comment about returning a default type passed if there's no type to parse sounds good and I can have something like `try_parse_type` for this, surely.

Reply:
@nunoplopes any thoughts on this? i have updated the patch with new `try_parse_type` method

Reply:
It doesn't seem to be needed. `try_parse_type` can simply tokenizer.peek() to see if there's a type following or not.
I'm just trying to intermingle these changes to the backbone with something else :)

Do you think the peek() solution works?

Reply:
Alright; peek solution works as well. I've updated the patch.